src/status_im/chat/models/message.cljs

```diff
@@ -97,11 +97,28 @@
 (defn add-outgoing-status [{:keys [from] :as message} {:keys [db]}]
   (assoc message :outgoing (= from (:current-public-key db))))
 
+(defn set-fiat-amount [{:keys [content-type content] :as message} cofx]
+  (if (= constants/content-type-command content-type)
+    (let [account-currency (-> cofx
+                               (get-in [:db :account/account :settings :wallet :currency])
+                               name
+                               string/upper-case)
+          {:keys [amount asset]} (:params content)
+          prices (get-in cofx [:db :prices])]
+      (-> message
+          (assoc-in [:content :params :fiat-amount] (money/fiat-amount-value amount
+                                                                             (keyword asset)
+                                                                             (keyword account-currency)
+                                                                             prices))
+          (assoc-in [:content :params :currency] account-currency)))
+    message))
+
```

Comment:
This is not a good place for the logic - we shouldn't pollute generic message handling with code specific to particular command, right place for it is `on-receive` protocol method of the `PersonalRequestCommand`.

Reply:
@janherich i thought about `on-receive`, but that call happens after message got saved to realm https://github.com/status-im/status-react/blob/c35a651f1599c76c0131e4af46eb9504b57ac10f/src/status_im/chat/models/message.cljs#L173 Do you think we can safely perform `on-receive` before `add-message-fn` ?

Reply:
@janherich or i could produce generate `data-store/tx` effect in `on-receive` and it will be then merged into all effects by `merge-fx`

Reply:
You are right, sorry for jumping the gun.
Let's do it the "simple way" now and in the meantime, I will think of something better, how generic receive flow can leverage `on-receive` protocol method in a more flexible way (I already have an idea).

Reply:
Sounds good. Also, i'd like to mention that my changes applied to both /receive and /send command, i.e. we calculate fiat amount on recipient side for both of them.